[GEOS-9696] Add documentation regarding JsonPointerFunction support (#4417)
geoserver/geoserver: doc/en/user/source/data/database/postgis.rst

   INSERT INTO place VALUES (ST_GeomFromText('POINT(-71.060316 48.432044)', 4326), E'Westfield\nTower');

JsonPointer Function support
----------------------------

> GeoServer is able to translate a jsonPointer function to a query using PostgreSQL support for JSON types. 

Reviewer: ```suggestion
GeoServer is able to translate the ``jsonPointer`` function to a query using PostgreSQL support for JSON types. 
```

Author: fixed

The following are the main characteristics of the implementation:

> * the jsonPointer function syntax is like the following: ``jsonPointer(attributeName,'/path/to/json/attribute')``;

Reviewer: Bullet lists in english are different than in italian, you should start each sentence with a capital and complete the sentence with a period, not a semicolon.

Author: fixed

While an example rule in a sld style sheet could be:

.. code-block:: xml 

   <Rule>
     <Name>Cities</Name>
        <ogc:Filter>
          <ogc:PropertyIsEqualTo>
            <ogc:Function name="jsonPointer">
              <ogc:PropertyName>city</ogc:PropertyName>
              <ogc:Literal>/population/average_age</ogc:Literal>
            </ogc:Function>
            <ogc:Literal>35</ogc:Literal>
          </ogc:PropertyIsEqualTo>
          </ogc:Filter>          
        <PointSymbolizer>
          <Graphic>
            <Mark>
              <WellKnownName>square</WellKnownName>
                <Fill>
                  <CssParameter name="fill">#FF0000</CssParameter>
                </Fill>
            </Mark>
            <Size>16</Size>
          </Graphic>
       </PointSymbolizer>
    </Rule>

DataTypes
`````````

PostgreSQL defines two JSON datatypes: 

 * ``json`` that stores an exact copy of the input text;

 * ``jsonb`` which store the value in a decomposed binary format.

The jsonPointer function supports both, as well as the text format if it contains a valid json representation. 

> Anyway it is worth mentioning that, according to the PostgreSQL documentation, jsonb is the recommended type to be used since it is faster to process. 

Reviewer: ```suggestion
Anyways, the the PostgreSQL documentation recommends usage of jsonb, as it's is faster to process. 
```

Author: fixed
